src/ray/core_worker/reference_count.cc

```diff
@@ -960,17 +960,33 @@ size_t ReferenceCounter::GetObjectSize(const ObjectID &object_id) const {
   return it->second.object_size;
 }
 
-void ReferenceCounter::HandleObjectSpilled(const ObjectID &object_id) {
+bool ReferenceCounter::HandleObjectSpilled(const ObjectID &object_id,
+                                           const std::string spilled_url,
+                                           const NodeID &spilled_node_id, int64_t size,
+                                           bool release) {
   absl::MutexLock lock(&mutex_);
   auto it = object_id_refs_.find(object_id);
   if (it == object_id_refs_.end()) {
     RAY_LOG(WARNING) << "Spilled object " << object_id << " already out of scope";
-    return;
+    return false;
   }
 
   it->second.spilled = true;
-  // Release the primary plasma copy, if any.
-  ReleasePlasmaObject(it);
+  if (spilled_url != "") {
+    it->second.spilled_url = spilled_url;
+  }
+  if (!spilled_node_id.IsNil()) {
+    it->second.spilled_node_id = spilled_node_id;
+  }
+  if (size > 0) {
```

Comment:
This if is not necessary right?

Reply:
I think it's necessary due to [this update](https://github.com/ray-project/ray/blob/e2d2c025e1092817c22c2d74bc5c7950ddfcd04b/src/ray/core_worker/core_worker.cc#L1320-L1321), which could overwrite a non-default object size.